src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -220,14 +215,12 @@ public function has($id)
             if (isset($this->aliases[$id])) {
                 $id = $this->aliases[$id];
             }
+            if (isset($this->privates[$id])) {
+                return false;
+            }
             if (isset($this->services[$id])) {
                 return true;
             }
-
-            if (isset($this->privates[$id])) {
```

Comment:
shouldn't this be moved earlier in 3.x ? If the private service was already instantiated, the code would return true above without triggering the notice